sdk/src/types/block/slot/epoch.rs

```diff
@@ -74,6 +74,12 @@ impl EpochIndex {
     pub fn last_slot_index(self, slots_per_epoch_exponent: u8) -> SlotIndex {
         SlotIndex::from_epoch_index(self + 1, slots_per_epoch_exponent) - 1
     }
+
+    /// Returns the slot at the end of which the validator and delegator registration ends and the voting power
+    /// for the epoch with index epoch + 1 is calculated.
+    pub fn registration_slot(&self, slots_per_epoch_exponent: u8, epoch_nearing_threshold: u32) -> SlotIndex {
+        self.last_slot_index(slots_per_epoch_exponent) - epoch_nearing_threshold
```

Comment:
True, I used last_slot_index because of https://github.com/iotaledger/iota.go/blob/42c1479d3f301ce7ca44871edebd4283727503ac/vm/nova/vm.go#L1042
Will ask in the TIP

Reply:
So in the TIP writes it's like this `StartSlotEpoch(n+1) - Nearing Threshold -1` and in iota.go and here (because I used iota.go as reference) we have `EndSlotEpoch(n) - NearingThreshold` which calculates actually the same thing `StartSlotEpoch(n+1) - 1 = EndEpochSlot(n)`